src/main/java/io/github/arrayv/groovyapi/RunSortBuilder.java

```diff
@@ -0,0 +1,273 @@
+            if (length < startingLength / 2) {
+                return defaultDelay * Math.pow(startingLength / 2048d, 2);
+            } else {
+                return defaultDelay * (startingLength / 2048d);
+            }
+        }
+        return defaultDelay;
+    }
+
+    private void removeClosers() {
+        if (Thread.currentThread() instanceof ScriptThread) {
+            ((ScriptThread)Thread.currentThread()).closers.remove(closer);
+        }
+        RunGroupContext rgc;
+        if ((rgc = RunGroupContext.CONTEXT.get()) != null) {
+            rgc.closers.remove(closer);
+        }
+        // Set this just in case we removed the wrong closer.
+        // This happens if we initialized this instance in a different thread
+        // from the one where we are calling finish().
+        closed = true;
+    }
+
+    private void finish() {
+        removeClosers();
+        if (RunGroupContext.CONTEXT.get() == null) {
+            final ArrayVisualizer arrayVisualizer = ArrayVisualizer.getInstance();
+            Thread sortThread = new Thread(this::run, "ScriptedSort");
+            arrayVisualizer.setSortingThread(sortThread);
+            arrayVisualizer.runSortingThread();
+            try {
+                sortThread.join();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        } else {
+            run();
+        }
+    }
+
+    private void run() {
+        final ArrayVisualizer arrayVisualizer = ArrayVisualizer.getInstance();
+        final ArrayManager arrayManager = arrayVisualizer.getArrayManager();
+        final ArrayFrame arrayFrame = arrayVisualizer.getArrayFrame();
+        final int[] array = arrayVisualizer.getArray();
+        final Delays Delays = arrayVisualizer.getDelays();
+        final Timer Timer = arrayVisualizer.getTimer();
+
+        final RunGroupContext runGroupContext = RunGroupContext.CONTEXT.get();
+        final int startingLength = runGroupContext != null ? runGroupContext.getStartingLength() : -1;
+
+        Delays.setSleepRatio(2.5);
+
+        int sortLength;
+        if (sort.isSlowSort()) {
+            sortLength = calculateLengthSlow(getLength(), sort.getUnreasonableLimit(), startingLength);
+        } else {
+            sortLength = calculateLength(getLength(), startingLength);
+        }
+        if (sortLength != arrayVisualizer.getCurrentLength()) {
+            arrayFrame.setLengthSlider(sortLength);
+        }
+
+        arrayManager.refreshArray(array, arrayVisualizer.getCurrentLength(), arrayVisualizer);
+
+        if (runGroupContext != null) {
+            arrayVisualizer.setHeading(sort.getRunAllName() + " (Sort " + runGroupContext.nextSort() + " of " + runGroupContext.getSortCount() + ")");
+        } else {
+            arrayVisualizer.setHeading(sort.getRunAllName());
+        }
+
+        double sortSpeed = calculateSpeed(getDelay(), arrayVisualizer.getCurrentLength(), startingLength);
```

Comment:
confusingly, "delay" is speed

Reply:
ArrayV has called it both in other places, so Idk what to do.

Reply:
just call it "speed" imo, "delay" is just wrong and it shouldn't conflict with anything

Reply:
I repeat, it's called delay in *so many* places in ArrayV. Not just here. Do you want me to change it everywhere? That's out of the scope of this PR.

Reply:
I never said that, and it's literally called <code>sort*Speed*</code> right there

I just mean changing it to "speed" within the groovy API

Reply:
That's fair